src/lib.rs

```diff
@@ -607,7 +630,7 @@ where
 /// assert_eq!(options.subsequent_indent, "  ");
 /// ```
 pub fn unfill(text: &str) -> (String, Options<'_>) {
-    let trimmed = text.trim_end_matches('\n');
+    let trimmed = text.trim_end_matches(&['\r', '\n']);
     let prefix_chars: &[_] = &[' ', '-', '+', '*', '>', '#', '/'];
 
     let mut options = Options::new(0);
```

Comment:
This comment should really be on the next line:

```rust
    for (idx, line) in trimmed.split('\n').enumerate() {
```

Just to double-check: splitting on `\n` still works because this also catches `\r\n`? You could consider moving the line ending detection logic up here and then use the detected line ending below. I think that should do the same, but it might be ever so slightly easier to understand.

Reply:
That's a good idea.

When working on this PR I wanted to minimize changes to the existing codebase, hence I wanted to leave the common prefix detection code intact (since my feature should not directly impact it). Hence the result. I tried to depict the data flow on the graph below

```mermaid
graph LR;
    input --> A & C
    subgraph loop [same loop]
      direction TB;
      C & B;
    end
    A[fold_prefix] --> B[fold_lines];
    C[fold_ending] --> D[Options];
    B --> E[&str];
    subgraph result [result]
      D & E
    end
```

We could of course do it like you ask

```mermaid
graph LR;
    input --> A & C
    subgraph loop [same loop]
      direction TB;
      A & C;
    end
    A[fold_prefix] --> B[fold_lines];
    C[fold_ending] --> D[Options];
    B --> E[&str];
    subgraph result [result]
      D & E
    end
```

We'll need new iterator to perform two fold in one pass, it'll contain all the logic (still the same complexity just moved around).
We'll still need two full passes over the input unfortunately, because we need the result of the `fold_prefix` before we do `fold_lines`. 

I'm sharing all this to prove that I've done my homework and thought about this :-) and with that in mind I'd prefer to leave the code as it is in this PR. I'd prefer this new feature to be as narrowly scoped as possible, and I'd be happy to start re-factoring in a new PR.

Reply:
That sounds good, I like it a lot that you try to keep the scope of the new feature narrow! It's hard to review PRs if they move random lines around and refactor the code while also making the actual changes :-)